.github/workflows/e2e_scripts/clean.sh

```diff
@@ -0,0 +1,101 @@
+#!/bin/bash
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -x
+set +e
+
+# This script is responsible for deleting resources out of a
+# stackdriver sandbox project in between test runs
+
+export PROJECT_ID=$(gcloud config get-value project)
+
+# delete cluster
+CLUSTER_ZONE="first_run"
+while [ -n "$CLUSTER_ZONE" ]; do
+  CLUSTER_ZONE=$(gcloud container clusters list \
+                   --filter="name:stackdriver-sandbox" \
+                   --project $PROJECT_ID --format="value(zone)")
+  if [ -n "$CLUSTER_ZONE" ]; then
+      echo "deleting cluster"
+      gcloud container clusters delete stackdriver-sandbox \
+          --project $PROJECT_ID --zone $CLUSTER_ZONE --quiet
+      sleep 20
+  fi
+done
+
+
+# delete loadgenerator
+LOADGENRATOR="first_run"
+while [ -n "$LOADGENRATOR" ]; do
+  read -r LOADGENRATOR ZONE <<< $(gcloud compute instances list \
+                                    --filter="name:loadgenerator*" \
+                                    --project $PROJECT_ID --format="value(name, zone)")
+  if [ -n "$LOADGENRATOR" ]; then
+      echo "deleting loadgenerator"
+      gcloud compute instances delete $LOADGENRATOR \
+          --project $PROJECT_ID --zone $ZONE --quiet
+      sleep 5
+  fi
+done
+
+# clear bucket
+BUCKET_CONTENTS="first_run"
+while [ -n "$BUCKET_CONTENTS" ]; do
+  BUCKET_CONTENTS=$(gsutil ls gs://$PROJECT_ID-bucket)
+  if [ -n "$BUCKET_CONTENTS" ]; then
+      echo "deleting bucket data"
+      gsutil rm -r "gs://$PROJECT_ID-bucket/**"
+      sleep 5
+  fi
+done
+
+# clear logs
+for LOG in $(gcloud logging logs list --project $PROJECT_ID --format="value(NAME)"); do
+    echo "deleting $LOG..."
+    gcloud logging logs delete $LOG --project $PROJECT_ID --quiet
+done
+
+# clear dashboards
```

Comment:
During Monitoring provisioning I also create the following
1) Custom Services
2) Uptime Check
I think these will also need to be cleaned up?

Reply:
Do you know if there are gcloud commands to manage those?

Reply:
Just checked - there isn't gcloud integration for either :(

Reply:
There is REST and client library support for both though

Reply:
Hmm ok, maybe we can add a curl command or something to handle that instead. Do you want to look into that as part of your monitoring tests later?

Reply:
Yeah that sounds good!

Reply:
Lack of gcloud for these is a big DevX gap, however it should be possible to destroy with ***terraform destroy***

Reply:
1. Will terraform destroy destroy the project as well?
2. I was hoping to avoid terraform destroy, because that relies on the existence of the tfstate file. I want to make sure `clean.sh` is reliable and idempotent, and not rely on existing state

Reply:
1) Currently monitoring terraform is separated so it will not destroy the project. however if we move them to be together then yes it would try to destroy the project.

Reply:
If there is no better solution currently available for removing all monitoring resources, it's better to use *terraform destroy* and not curl-based API calls but anyway, this doesn't have to be part of this PR